src/test/PulsarMock.js

```diff
@@ -0,0 +1,251 @@
+export class PulsarClientMock {
+  // A map used to simulate an in-memory store for messages. The keys are topic names, and the values are arrays of message objects.
+  static inMemoryStore = new Map();
+
+  // A counter to generate unique message IDs. It is incremented each time a message is sent.
+  static messageIdCounter = 0;
+
+  // A map where keys are topic names and values are arrays of receive function invocations (listeners). These listeners are called to notify about new messages.
+  static listeners = new Map();
+
+  // The default time in milliseconds to wait before a message is considered not acknowledged (acknowledged messages are removed from visibility to simulate message acknowledgment behavior).
```

Comment:
why do ack's need timeouts? Is that something pulsar-like that we are trying to simulate? What is the concept of "visibility"?